R/execute-data-diffs.R

```diff
@@ -181,22 +181,36 @@ execute_data_diffs <- function(.base_df, .compare_df, .subject_col, .base_from_s
       .compare_df[.compare_df[[.subject_col]] == id.i, ] %>%
       dplyr::select(dplyr::all_of(names_in_common))
 
-    diff.i <-
-      suppressMessages(
-        diffdf::diffdf(
-          base = base_df.i,
-          compare = compare_df.i,
-          suppress_warnings = TRUE,
-          strict_numeric = FALSE,
-          strict_factor = FALSE
+    # If they are identical (we do not care if attributes are different),
+    # skip the diffdf check to save time
+    all_equal.i <- all.equal(
+      target = base_df.i,
+      current = compare_df.i,
+      check.attributes = FALSE
+    )
+
+    if (!isTRUE(all_equal.i)) {
+
+      diff.i <-
+        suppressMessages(
+          diffdf::diffdf(
+            base = base_df.i,
+            compare = compare_df.i,
+            suppress_warnings = TRUE,
+            strict_numeric = FALSE,
+            strict_factor = FALSE
+          )
```

Comment:
I like this change. Do we want a test for this or no?

Reply:
@barrettk A test would be helpful tbh